Modifies GET user_role endpoint

dxenes1 opened the pull request:
Nicole needed a way to get a logged-in user's roles without needing an admin account. However, the issue is that the GET user_roles endpoint requires "user-manager" role which blocks us from authorizing users to make edits to the metadata site.  

This change drop the "user-manager" requirement on the GET user_roles endpoint and allow anyone to request a list of their own roles for their account. "user-manager" role is required if you want to know the role of other usernames still.

dxenes1 committed:
modifies GET user_role endpoint

dxenes1 committed:
update username in tests

dxenes1 committed:
adds new test for self role identity

movestill commented on django/sso/views/views_user.py:
```diff
@@ -236,6 +245,15 @@ def get(self, request, user_name, role_name=None):
+                if user_name is None:
+                    user_name = signed_in_user
+                else:
+                    resp = check_for_user_manager(signed_in_user)
```
I think this might read better if you raise an exception in `check_for_user_manager()` and catch it down below. Then in the handler, return a `BossHTTPError`.

movestill committed:
Fix and improve user unit tests

Create users and their roles in the test DB so requests sent with "real"
users.  Remove fake of `check_roles()` since we're now using "real"
users.

movestill reviewed (commented):
Didn't realize that I never finished my review!
